### PR TITLE
Bootstrap and DataTables version upgrade

### PR DESCRIPTION
**Pull request by AlbertWhitlock**
Bootstrap version is now 5.0.2 and DataTables version is now 1.12.1.

### CODE REVIEW

**Comment by DomGarguilo** [comment]
I'll take a look into these changes today

**Comment by DomGarguilo** [comment]
I took some screenshots on main and this branch and noticed some differences. Some changes can probably stay how they are, others we might want to correct.

**On the home page:**

main:
Screenshot from 2022-08-04 11-00-23

this branch:
Screenshot from 2022-08-04 10-25-50

The overview table should probably be returned to center. Also, there is a white bar at the top of every page for some reason on this branch that should probably be removed.

**On the manager page:**

main:
Screenshot from 2022-08-04 11-00-54

this branch:
Screenshot from 2022-08-04 10-26-11


It looks like the table titles appear below the table which should probably be moved back above the table. This might be the new default for DataTables, but it might be better if the titles are at the top of the tables.

**Dropdown menus:**

main:
Screenshot from 2022-08-04 11-01-14

this branch:
Screenshot from 2022-08-04 10-26-25

It looks like the text for the menu options is running outside of its background on this branch. Also, the button that was clicked to bring down this dropdown menu (`Servers` in this case) is not highlighted anymore.

**The info modal:**

main:
Screenshot from 2022-08-04 11-02-01

this branch:
Screenshot from 2022-08-04 10-27-13

On this branch the `X` button in the top right was removed and the close button was replaced by an `X` button. Personally I think this looks good but it might be good to make the modal a bit wider so that the Instance ID does not have to wrap to the next line.

**The navbar:**

main:
Screenshot from 2022-08-04 11-02-29

this branch:
Screenshot from 2022-08-04 10-27-54

I see two main differences that both seem like they should be returned to how they previously were, if possible. The counter next to `debug` is no longer highlighted, and the 3 dots button has been changed to a help button. Also it looks like there used to be a gradient for the navbar instead of a solid color like it is on this branch.

**Some of the bottons:**

main:
Screenshot from 2022-08-04 11-03-00

this branch:
Screenshot from 2022-08-04 10-31-03

Some of the styling for the buttons has changed. I personally like the look of the old button styling (with the white text and gradient background) better but not a big deal.

Here are some actionable points that can be checked off as they are completed or otherwise addressed:

- [ ] center the overview table on the home page
- [ ] remove the white bar at the top of the pages
- [ ] move the table titles back to the top of the tables
- [ ] make sure the text in the drop down menus on the nav bar are not off their background
- [ ] make sure the button clicked on the nav bar are highlighted while selected
- [ ] make the info modal wider so that the instance ID doesn't have to wrap
- [ ] highlight the counter next to debug on the nav bar
- [ ] change the help button on the nav bar back to the 3 dots
- [ ] consider reverting the styling for the buttons/navbar having a gradient background instead of solid

**Comment by DomGarguilo** [comment]
Looks like just about everything I had mentioned has been addressed. I updated the checklist above. I found a few more things that might need changing:

It doesn't look like the auto-refresh is working anymore

It looks like at least one of the drop down menus is cut off:
Screenshot from 2022-08-11 16-50-49

Also when opening this option, it covers part of the nav bar (and the botton is not highlighted when clicked like the others):
Screenshot from 2022-08-11 16-50-52

I wonder if there is a way to have these two display the drop down portion with the right side aligned (rather than the usual left side of the button aligned with the left side of the dropdown).

Other than those things, everything else looks good as far as I have seen.

**Comment by DomGarguilo** [comment]
357502a resolved my concerns about the drop downs (mentioned above).

Updated dropdowns screenshots:
Screenshot from 2022-08-12 14-35-58
Screenshot from 2022-08-12 14-35-50

**Comment by Manno15** [comment]
I did run into an interesting bug. I had the monitor auto-refreshing but none of the drop-downs worked. The buttons themselves worked but that is about it. I think this is because I had the auto-refresh enabled on the monitor before updating to these changes.

**Comment by DomGarguilo** [comment]
I took a look at 2ce1953 and it looks like it fixed the auto-refresh feature. While testing it out though, I noticed something else. When a something is present in the debug menu, the counter stays green...
Screenshot from 2022-08-23 15-11-09
Screenshot from 2022-08-23 15-11-19
until the recent logs page is visited at which-point the counter turns to the yellow warning color as is expected whenever that counter is not zero.
Screenshot from 2022-08-23 15-11-36

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/scans.ftl [hunk 1]
```diff
@@ -19,13 +19,14 @@
 
 -->
       <div class="row">
-        <div class="col-xs-12">
-          <h3>${title}</h3>
-        </div>
+         <div class="col-xs-12">
+            <h3>Active Scans Overview</h3>
+         </div>
```
Comment (DomGarguilo) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/scans.ftl:24]: Looks like a few extra spaces got added here, might be best to remove them.

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/tables.ftl [hunk 2]
```diff
@@ -87,9 +87,14 @@
...
         }
       </script>
+      <div class="row">
+         <div class="col-xs-12">
+            <h3>Table Overview</h3>
+         </div>
+      </div>
       <div>
-        <table id="tableList" class="table table-bordered table-striped table-condensed">
-          <caption><span class="table-caption">${tablesTitle}</span><br />
```
Comment (DomGarguilo) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/tables.ftl:92]: Why were all of the field substitutions (`${}`) replaced with hardcoded values?

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/tservers.ftl [hunk 3]
```diff
@@ -18,39 +18,15 @@
...
 
 -->
-      <div class="row">
+    <div class="row">
         <div class="col-xs-12">
-          <h3>${title}</h3>
+            <h3>Tablet Server Overview</h3>
         </div>
-      </div>
+    </div>
```
Comment (DomGarguilo) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/tservers.ftl:25]: looks like this div is out of line with the others. also the `<h3>` has one or two extra spaces added.

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/scans.ftl (hunk 1 above)
Comment (DomGarguilo) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/scans.ftl:24]: Looks like some extra spaces got added on these lines

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/navbar.ftl [hunk 4]
```diff
@@ -18,73 +18,71 @@
...
-              </a>
-              <ul class="dropdown-menu">
-                <li><a class="auto-refresh" style="cursor:pointer">Auto-Refresh</a></li>
-                <li><a data-toggle="modal" href="#aboutModal">About</a></li>
-              </ul>
-            </li>
-          </ul>
-        </div>
-      </div>
+<nav class="navbar navbar-expand-lg navbar-dark bg-dark">
```
Comment (DomGarguilo) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/navbar.ftl:21]: Not sure what happened, but it looks like the formatting on this whole file got messed up somehow- should probably be returned to proper formatting.

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/navbar.ftl [hunk 5]
```diff
@@ -18,73 +18,71 @@
...
+  <div class="container-fluid">
+    <a class="navbar-brand" id="headertitle" href="/">
+                <img id="accumulo-avatar" alt="accumulo" class="navbar-left pull-left" src="/resources/images/accumulo-avatar.png" />
+                uno
```
Comment (DomGarguilo) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/navbar.ftl:25]: The instance name will not always be uno. Thats why there was the `${instance_name}` that gets replaces with the instance name dynamically

**Review by DomGarguilo — CHANGES_REQUESTED** [review]

**Comment by Manno15** [comment]
Recent commits have solved my issues so once @DomGarguilo's issues are resolved then this should be good to go. Good work.

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/navbar.ftl (hunk 5 above)
Comment (EdColeman) [reply server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/navbar.ftl:25]: There could be value in also trying running this in a small cluster (could use muchos) to check that it is not relying on uno / mini-cluster specific configuration / conventions that would not be valid in a cluster / production environment.   (Triggered by @DomGarguilo comment that the name might not always be uno)

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/tables.ftl (hunk 2 above)
Comment (AlbertWhitlock) [reply server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/tables.ftl:92]: I put captions on each table and sometimes the 'title' of the page (that was at the top of the page) worked as well for a table caption so I added a different, hardcoded value for the page's overall title. I put the field substitution values back for the pages' titles and just changed the individual tables' captions.

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/navbar.ftl (hunk 5 above)
Comment (AlbertWhitlock) [reply server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/navbar.ftl]: Changed it back to ${instance_name}

**Review by DomGarguilo — APPROVED** [review]
I think all my concerns have been resolved. This looks good to me.

**Comment by ctubbsii** [comment]
@DomGarguilo Please don't forget to ensure the version-specific project is added before merging PRs

**Pull request merged**